versioned_docs/version-2.0/tutorials/backtesting.mdx

```diff
@@ -0,0 +1,185 @@
+    """Chat model response metadata @LangChainAI chat model invocations now include metadata like logprobs directly in the output. Upgrade your version of `langchain-core` to try it. PY: https://python.langchain.com/docs/modules/model_io/chat/logprobs JS: https://js.langchain.com/docs/integrations/chat/openai#generation-metadata""",
+    """Benchmarking Query Analysis in High Cardinality Situations Handling high-cardinality categorical values can be challenging. This blog explores 6 different approaches you can take in these situations. Blog: https://blog.langchain.dev/high-cardinality""",
+    """Building Google's Dramatron with LangGraph.js & Claude 3 We just released a long YouTube video (1.5 hours!) on building Dramatron using LangGraphJS and @AnthropicAI's Claude 3 "Haiku" model. It's a perfect fit for LangGraph.js and Haiku's speed. Check out the tutorial: https://youtube.com/watch?v=alHnQjyn7hg""",
+    """Document Loading Webinar with @AirbyteHQ Join a webinar on document loading with PyAirbyte and LangChain on 3/14 at 10am PDT. Features our founding engineer @eyfriis and the @aaronsteers and Bindi Pankhudi team. Register: https://airbyte.com/session/airbyte-monthly-ai-demo""",
+]
+
+_ = system.batch(
+    [{"messages": [HumanMessage(content=content)]} for content in inputs],
+    {"max_concurrency": 3},
+)
+```
+
+## Convert Prod Runs to Test
+
+The first step is to generate a dataset based on the production _inputs_.
+Then copy over all the traces to serve as a baseline run.
+
+```python
+from datetime import datetime, timedelta, timezone
+
+from langsmith import Client
+from langsmith.beta import convert_runs_to_test
+
+# How we are sampling runs to include in our test
+end_time = datetime.now(tz=timezone.utc)
+start_time = end_time - timedelta(days=1)
+run_filter = f'and(gt(start_time, "{start_time.isoformat()}"), lt(end_time, "{end_time.isoformat()}"))'
+
+
+# Fetch the runs we want to convert to a test
+client = Client()
+
+prod_runs = list(
+    client.list_runs(
+        project_name=project_name,
+        execution_order=1,
+        filter=run_filter,
+    )
+)
+
+# Name of the dataset we want to create
+dataset_name = f'{project_name}-backtesting {start_time.strftime("%Y-%m-%d")}-{end_time.strftime("%Y-%m-%d")}'
+# This converts the runs to a dataset + test
+# It does not actually invoke your model
+convert_runs_to_test(
+    prod_runs,
+    # Name of the resulting dataset
+    dataset_name=dataset_name,
+    # Whether to include the run outputs as reference/ground truth
+    include_outputs=False,
+    # Whether to include the full traces in the resulting test project
+    # (default is to just include the root run)
+    load_child_runs=True,
+)
+```
```

Comment:
Do we even want to show this output here or should we just remove? Just shows the beta warning and a bunch of data that isn't very useful to the user imo.